### PR TITLE
Refactor appointments API and frontend into smaller segments

### PR DESCRIPTION
**Pull request by Tom-Goring**
This PR folds the appointments frontend into other pages and seperates out the dashboard dispatchers into their own spaces. 

- The appointments JSP has been split into create and update - these will need styling
- The staff and patient dashboards have had appointment tables added - these will also need styling.
- A new getAppointmentByID() method was added to the AppointmentDAO
- New servlets for creating, updating, and deleting appointments have been added alongside corresponding endpoints (declared using annotations)

### CODE REVIEW

File: src/main/java/Controllers/Appointments/Update.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        request.getRequestDispatcher("/appointments/update.jsp").forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Integer appointmentID = Integer.parseInt(request.getParameter("appointmentID"));
+        Appointment appointment = AppointmentDAO.getAppointmentByID(appointmentID);
+
+        String type = request.getParameter("serviceType");
+        assert appointment != null; // god save my soul
```
Comment (CharlievWilliams) [inline src/main/java/Controllers/Appointments/Update.java:27]: Maybe remove this if the lecturers plan to read our code lol

File: src/main/java/Filters/DashboardFilter.java [hunk 2]
```diff
@@ -6,242 +6,50 @@
...
 
 @WebFilter(filterName = "DashboardFilter")
 public class DashboardFilter implements Filter {
 
+    private final String[] pages = {"appointments", "patientinformation"};
+
     @Override
-    public void destroy() {
-    }
+    public void destroy() {}
```
Comment (CharlievWilliams) [inline src/main/java/Filters/DashboardFilter.java:22]: You and my autoformat appear to be battling it out here lmao

File: src/main/java/Models/Appointment/Appointment.java [hunk 3]
```diff
@@ -61,6 +63,18 @@ public String getType() {
         return type;
     }
 
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public static String getDayOfDate(Date date) {
```
Comment (CharlievWilliams) [inline src/main/java/Models/Appointment/Appointment.java:70]: This is a reminder to myself to update all Appointment unit tests, as they will all be broken now.

**Review by CharlievWilliams — APPROVED** [review]

File: src/main/java/Filters/DashboardFilter.java (hunk 2 above)
Comment (Tom-Goring) [reply src/main/java/Filters/DashboardFilter.java:22]: You can change that in the Intellij settings - both are okay in the java formatting standards I think. I prefer single line empty to multiline

Comment (Tom-Goring) [reply src/main/java/Filters/DashboardFilter.java:22]: But I'm just saying mine's gonna win

File: src/main/java/Controllers/Appointments/Update.java (hunk 1 above)
Comment (Tom-Goring) [reply src/main/java/Controllers/Appointments/Update.java:27]: It will be visible forever in the commit history, my mark has been left

**Review by Maxwell2Gyamfi — APPROVED** [review]

**Pull request merged**